File: 4 Kyu/4 Kyu - Magnet particules in boxes.c
#include <math.h>

double v (int k, int n) {
    return 1 / (k * pow ((n + 1), 2 * k));
}
double u (int k, int N) {
    double sum = 0; for (int n = 1; n <= N; ++n) sum += v (k,n);
    return sum;
}
double doubles (int maxk, int maxn) {
    double sum = 0; for (int k = 1; k <= maxk; ++k) sum += u (k,maxn);
    return sum;
}

File: 4 Kyu/4 Kyu - Magnet particules in boxes.rs
fn doubles (maxk: i32, maxn: i32) -> f64 {
    let mut sum: f64 = 0.0;

    for k in 1..= maxk {
        for n in 1..= maxn {
            sum += 1.0 / (k as f64 * f64::powf (n as f64 + 1.0, 2.0 * k as f64));
        }
    }
    
    sum 
}
